pkg/gadgets/trace/network/types/types.go

```diff
@@ -54,23 +47,21 @@ func (e *Event) SetLocalPodDetails(owner, hostIP, podIP string, labels map[strin
 	e.PodLabels = labels
 }
 
-func (e *Event) GetRemoteIPs() []string {
-	return []string{e.RemoteAddr}
+func (e *Event) GetEndpoints() []eventtypes.Endpoint {
+	return []eventtypes.Endpoint{e.DstEndpoint}
 }
 
-func (e *Event) SetEndpointsDetails(endpoints []eventtypes.EndpointDetails) {
+func (e *Event) SetEndpoints(endpoints []eventtypes.Endpoint) {
 	if len(endpoints) == 0 {
 		return
 	}
-	e.RemoteName = endpoints[0].Name
-	e.RemoteNamespace = endpoints[0].Namespace
-	e.RemoteLabels = endpoints[0].PodLabels
-	e.RemoteKind = endpoints[0].Kind
+	e.DstEndpoint = endpoints[0]
 }
 
 func GetColumns() *columns.Columns[Event] {
-	cols := columns.MustCreateColumns[Event]()
+	cols := columns.MustCreateColumns[Event](columns.WithPrependEmbeddedNames(true))
 
+	// Can't use eventtypes.AddVirtualColumn here because we don't capture port information and dont' want to display it
 	cols.MustAddColumn(columns.Attributes{
```

Comment:
Shouldn't it use `eventtypes.AddVirtualColumn` too?

Reply:
The printing is here done differently. Here the port is not displayed and not captured at all. The output would always show a port `1.1.1.1:0` or `127.0.0.1:0`

Reply:
Theoretically port 0 is allowed and could be used by someone (Even if it is not seen good). So i guess a pointer would be needed here?

Regarding the difference of `MustAddVirtualEndpointColumn` and the formatting in `trace network`: I left the network formatting untouched so the output wouldn't change at all. But I don't have anything against aligning it

Reply:
This one is difficult to handle. IMO we should consider the case of port 0 being valid. I also think adding a pointer creates some complications for the user of those types, as they need to check if this is nil before reading and when assigning it has to be saved in a variable in order to be able to get its address.

I think a solution is to split Endpoint into two types: L3 (IP) and L4 (ports). I did exploration in https://github.com/inspektor-gadget/inspektor-gadget/commit/12e216a59509f754c3c7de93ed6d624a180fdde4 and I think the result is good. What do you think?

Reply:
Looks nice @mauriciovasquezbernal , thanks. I cherry-picked it